### ISSUE TITLE
Laser Mirror

### ISSUE THREAD

**Issue by CrashingScience** [TeamSpen210/BEE2-items#576]
I mentioned this earlier, but it didn't really get noticed. Glass which is a mirror on one side, reflect lasers on that side and letting them pass through on the other. Start reversed changes the reflected side.

**Comment by TeamSpen210** [comment]
I don't think this is possible to implement.

**Comment by BenVlodgi** [comment]
It's probably possible, the mirror part would be too hefty on the engine though, so just a reflective surface on the mirror side.

**Comment by CrashingScience** [comment]
Yes, i know mirrors are processor intensive, i meant just shiny-looking

**Comment by BenVlodgi** [comment]
Shouldn't be too difficult really
reflection code already exists for the gel
would love to see an example

**Comment by TeamSpen210** [comment]
The code does exist, the problem is that there isn't any way to hook into it.

**Comment by BenVlodgi** [comment]
could cheat by painting the surface with reflectiongel
and if you wanted to give it a different look, you could have the brush that it being painted be 0.25 units (or fewer) behind another brush that has the shiny mirror texture

**Comment by TeamSpen210** [comment]
That could be done, the problem is that non-solid or transparent surfaces aren't paintable. It'd block lasers passing from the other side.

**Comment by BenVlodgi** [comment]
The non-solid surface that sticks out fractions of a unit would just be for appearances... you could have a metal texture behind it (which would be painted)

**Comment by ellingtonvengaboysmuseum** [comment]
this could be done with a custom fizzler with one surface painted with reflection gel with rendermode 10 to make it invisible.

**Comment by BenVlodgi** [comment]
> custom fizzler with one surface painted with reflection gel with rendermode 10

func_brush has a rendermode property too

I built a prototype, it reflects off of the mirror side, and you can see through the other side. But you can't send a laser through the glass side.
In the end, there is pretty much no point in having it as a one-way mirror, might as well just have it as a 2-sided mirror.

**Comment by Luke18033** [comment]
How about a resizeable 2-way mirror?

**Comment by TeamSpen210** [comment]
That's doable. In reality, mirrors aren't really one-way - on both sides they reflect and transfer light. The 'hidden' side is kept dark, so it's not very easy to see inside. It's easiest to just make a regular mirror item, or alternatively a mirror with black tiles on the other side.

**Comment by BenVlodgi** [comment]
true story
a small problem is the reflective part has to be pained, but I want to avoid the paint from splashing on the map
thinking about teleporting the paintable brush out of the map, paint it, and teleport it back

**Comment by CrashingScience** [comment]
Thats cool too

**Comment by BenVlodgi** [comment]
I streamed earlier and did show what I had done with this item. [Here is when.](https://youtu.be/Ge18ahKJ6mw?t=100)

**Comment by LautaroL20** [comment]
Im not sure, but this item could end up detracting the Reflection Gel test element.

**Comment by LautaroL20** [comment]
This item definitelly detracts the noble Reflection Gel element, im even working on a test chamber with it. Case closed.

**Comment by BenVlodgi** [comment]
Most new items detract from other mechanics.
Reflection gel detracts from Portals and laser redirection cubes
Mirrors "could" be a good way to ensure that the mirror element is always present in the puzzle.
Reflection gel can be washed away.
I'm not saying that mirrors have to be added, but I don't think there is any reason to close this issue.

**Comment by LautaroL20** [comment]
Ok, everyone is entitled to have different point of views. As for redirection cubes, i don´t think Reflection Gel would detract them since you can combine Non-Portalable Walls - Angled Panes that have been painted with this reflective gel.

**Comment by TeamSpen210** [comment]
I'll be implementing Felix Griffin's [mirror devices](http://steamcommunity.com/sharedfiles/filedetails/?id=122072276) for this. They aren't a thin wall, but they do actually reflect lasers, funnels, bridges and pellets. (Basically they're a pair of invisible portals with a non-solid wall in front of them, and clips to stop physics objects.)

**Comment by LautaroL20** [comment]
It will reflect Light Bridges? Well, that sounds much better then, since reflection gel actually only reflects lasers.

**Comment by TeamSpen210** [comment]
See [this view](http://steamcommunity.com/sharedfiles/filedetails/?id=122031132) to see how it works internally - it should be able to handle basically everything.

**Comment by CrashingScience** [comment]
Looks really cool, being able to redirect all sorts of things.
